modules/data.atmosphere/R/tdm_temporal_downscale_functions.R

```diff
@@ -0,0 +1,1134 @@
+            nc <- ncdf4::nc_create(outfile, var.list)
+            ncdf4::ncvar_put(nc, var.list, mod.out[[i]][["betas"]])
+            ncdf4::nc_close(nc)
+            
+            # Save the model as a .Rdata
+            mod.save <- mod.out[[i]][["model"]]
+            save(mod.save, file = file.path(path.out, paste0("model_air_temperature_", 
+                i, ".Rdata")))
+        }
+    } else {
+        for (i in names(dat.list)) {
+            mod.out <- model.train(dat.subset = dat.list[[i]], n.beta = n.beta, 
+                resids = resids)
+            
+            # Save the betas as .nc
+            outfile <- file.path(path.out, paste0("betas_air_temperature_", 
+                i, ".nc"))
+            dimY <- ncdf4::ncdim_def(paste0("coeffs_", i), units = "unitless", 
+                longname = "model.out coefficients", vals = 1:ncol(mod.out[["betas"]]))
+            dimX <- ncdf4::ncdim_def("random", units = "unitless", longname = "random betas", 
+                vals = 1:nrow(mod.out[["betas"]]))
+            var.list <- ncdf4::ncvar_def(i, units = "coefficients", dim = list(dimX, 
+                dimY), longname = paste0("day ", i, " model.out coefficients"))
+            nc <- ncdf4::nc_create(outfile, var.list)
+            ncdf4::ncvar_put(nc, var.list, mod.out[["betas"]])
+            ncdf4::nc_close(nc)
+            
+            # Save the model as a .Rdata
+            mod.save <- mod.out$mode
+            save(mod.save, file = file.path(path.out, paste0("model_air_temperature_", 
+                i, ".Rdata")))
+        }
+    }
+    # return(mod.out)
+}
+
+model.surface_downwelling_shortwave_flux_in_air <- function(dat.train, 
+    n.beta, path.out, resids = F, parallel = F, n.cores = NULL, day.window = 5, 
+    seed = 1341) {
+    
+    set.seed(seed)
+    if (!dir.exists(path.out)) 
+        dir.create(path.out, recursive = T)
+    # The model we're going to use
+    model.train <- function(dat.subset, threshold = NULL, n.beta, resids = resids) {
+        
+        # Don't bother trying to fit hours that are completely or pretty darn
+        # close to dark
+        hrs.day <- unique(dat.subset[dat.subset$surface_downwelling_shortwave_flux_in_air > 
+            threshold, "hour"])
+        
+        # Note: played around with a log-transformation of
+        # surface_downwelling_shortwave_flux_in_air to prevent negative values,
+        # but that resulted in bias at upper range Solution was to just say
+        # anything <0 = 0 mod.doy <-
+        # lm(surface_downwelling_shortwave_flux_in_air ~
+        # as.factor(hour)*surface_downwelling_shortwave_flux_in_air.day,
+        # data=dat.subset[dat.subset$hour %in% hrs.day,]) ###
+        mod.doy <- lm(surface_downwelling_shortwave_flux_in_air ~ as.factor(hour) * 
+            surface_downwelling_shortwave_flux_in_air.day - 1 - surface_downwelling_shortwave_flux_in_air.day - 
+            as.factor(hour), data = dat.subset[dat.subset$hour %in% hrs.day, 
```

Comment:
I didn't comment on this with the first model (air temp), since I assumed that was the starting point, of the model building, but don't later variable need to also depend on earlier variables if we want to capture the covariance among variables? (i.e. why isn't the instantaneous Tair a covariate here?)